src/models/spaces/organisms/organisms-mouse.ts

```diff
@@ -42,108 +41,30 @@ const SubstanceDelta = types
-        kSubstanceNames.forEach((substance: SubstanceType) => {
-          self.matrix.get(organelle)!.set(substance, SubstanceDelta.create());
-        });
-      });
-    },
-
-    addSubstanceToOrganelle(organelle: OrganelleType, substance: SubstanceType, inc: number, decayStart?: number) {
-      const delta = self.getDeltaForOrganelleSubstance(organelle, substance);
-      delta.setAmount(delta.amount + inc);
-
-      if (decayStart) {
-        delta.setDecayStart(decayStart);
-      }
-    },
-
-    step(
-      newTime: number,
-      extracellularHormone: number,
-      cytoplasmProtein: number,
-      melanosomeEumelanin: number,
-      melanosomePheomelanin: number)
-    {
-      kOrganelleNames.forEach((organelle: OrganelleType) => {
-        kSubstanceNames.forEach((substance: SubstanceType) => {
-          const delta = self.getDeltaForOrganelleSubstance(organelle, substance);
-
-          if (newTime < delta.decayStart) {
-            return;
-          }
-
-          let birthRate;
-          let deathRate;
-          switch (substance) {
-            case "hormone":
-              birthRate = organelle === "extracellular"
-                ? 300 / 20
-                : 0;
-              deathRate = (100 + 1.6 * extracellularHormone) / 20;
-              break;
-            case "signalProtein":
-              birthRate = organelle === "cytoplasm"
-                ? self.organismColor === "brown"
-                  ? (180 + .8 * extracellularHormone) / 10
-                  : self.organismColor === "tan"
-                    ? (90 + .8 * extracellularHormone) / 10
-                    : 25 / 10
-                : 0;
-              deathRate = (25 + 1.5 * cytoplasmProtein) / 10;
-              break;
-            case "eumelanin":
-              birthRate = organelle === "melanosome"
-                ? self.organismColor !== "white"
-                  ? (280 + 1.5 * cytoplasmProtein) / 10
-                  : (25 + 1.8 * cytoplasmProtein) / 10
-                : 0;
-              deathRate = (25 + 1.5 * melanosomeEumelanin) / 10;
-              break;
-            case "pheomelanin":
-              birthRate = organelle === "melanosome"
-                ? (500 - 1.5 * cytoplasmProtein) / 10
-                : 0;
-              deathRate = (25 + 1.5 * melanosomePheomelanin) / 10;
-              break;
-            default:
-              birthRate = 0;
-              deathRate = 0;
-              break;
-          }
-
-          delta.setAmount(delta.amount + birthRate - deathRate);
-        });
-      });
-    }
-  }));
-
+const kAssayableOrganelles: OrganelleType[] = ["extracellular", "cytoplasm", "melanosome"];
 export const OrganismsMouseModel = types
   .model("OrganismsMouse", {
     id: types.optional(types.identifier, () => uuid()),
     backpackMouse: types.reference(BackpackMouse),
-    substanceDeltas: types.maybe(OrganelleSubstanceDeltas),
+    substanceDeltas: types.map(types.map(SubstanceDelta)),
```

Comment:
I ended up putting `substanceDeltas` directly in the mouse, rather than a separate model